### PR TITLE
Set recovery rate for dedicated cold nodes

### LINKED ISSUE #68134: Ergonomic settings for indices recovery based on node size for data_cold role
To enhance the performance of [the cold tier](https://www.elastic.co/guide/en/elasticsearch/reference/7.11/data-tiers.html#cold-tier) during failure scenarios or prewarming we could move away from the fixed `40MB/s` value for [indices.recovery/max_bytes_per_sec](https://www.elastic.co/guide/en/elasticsearch/reference/7.11/recovery.html#_recovery_settings) and instead ergonomically choose the value based on the node size.

So for nodes with the `data_cold` [node_role](https://github.com/elastic/elasticsearch/pull/60994) we could implement a scheme like:

**NOTE**: the following table is referring to the available memory for the node (e.g. memory available inside a container) not jvm heap size

| node size in GB | indices.recovery.max_bytes_per_sec | change from current default |
| --------------------------- | ------------------------------------------------- | --- |
| node_size <= 1 | 40 MB/s | unchanged | 
| 1 < node_size <= 2 | 40 MB/s | unchanged |
| 2 < node_size  <= 4 | 40 MB/s | unchanged |
| 4 < node_size <= 8 | 60 MB/s | increased |
| 8 < node_size <= 16 | 90 MB/s | increased |
| 16 < node_size <= 32 | 125 MB/s | increased |
| 32 < node_size | 250 MB/s | increased |

### PR DESCRIPTION
**Pull request by jasontedor**
This commit sets the recovery rate for dedicated cold nodes. The goal is here is enhance performance of recovery in a dedicated cold tier, where we expect such nodes to be predominantly using searchable snapshots to back the indices located on them. This commit follows a simple approach where we increase the recovery rate as a function of the node size, for nodes that appear to be dedicated cold nodes.

Closes #68134

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java [hunk 1]
```diff
@@ -12,21 +12,64 @@
...
+            s -> {
+                final ByteSizeValue defaultMaxBytesPerSec = new ByteSizeValue(40, ByteSizeUnit.MB);
+                final List<DiscoveryNodeRole> roles = NodeRoleSettings.NODE_ROLES_SETTING.get(s);
+                final List<DiscoveryNodeRole> dataRoles =
+                    roles.stream().filter(DiscoveryNodeRole::canContainData).collect(Collectors.toUnmodifiableList());
+                if (dataRoles.isEmpty()) {
+                    // if the node is not a data node, this value doesn't matter, use the default
+                    return defaultMaxBytesPerSec.getStringRep();
+                }
+                if ((dataRoles.size() > 1 || dataRoles.get(0).roleName().equals("data_cold") == false) ||
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java:45]: Note: we should bring the role definitions into the server code base now.

**Review by dakrone — APPROVED** [review]
LGTM, should we also change the docs for `indices.recovery.max_bytes_per_sec` to reflect this?

**Comment by jasontedor** [comment]
@dakrone I pushed docs.

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Review by dakrone — APPROVED** [review]
Docs look good to me also :+1:

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Pull request merged**